source/jormungandr/jormungandr/scenarios/experimental.py

```diff
@@ -595,172 +51,97 @@ def call_kraken(self, request_type, request, instance, krakens_call):
-
-            if local_resp.HasField(b"error") and local_resp.error.id == response_pb2.Error.error_id.Value('no_solution') \
-                    and direct_path.journeys:
-                local_resp.ClearField(b"error")
-            if local_resp.HasField(b"error"):
-                #Here needs to modify error message of no_solution
-                if len(g.origins_fallback[dep_mode]) == 0:
-                    self.update_error_message(local_resp, response_pb2.Error.no_origin, "no origin point")
-                elif len(g.destinations_fallback[arr_mode]) == 0:
-                    self.update_error_message(local_resp, response_pb2.Error.no_destination, "no destination point")
-
-                return [local_resp]
-
-            # for log purpose we put and id in each journeys
-            for j in local_resp.journeys:
-                j.internal_id = str(generate_id())
-                response_tuples.append((dep_mode, arr_mode, j))
-            resp.append(local_resp)
-
-        # Once the pt journey is found, we need to reconstruct the whole journey with fallback regarding the mode
-
-        # For the sake of performance, we compute at first all fallback direct path asynchronously
-        # then we update the pool of direct paths
-        futures = worker.get_fallback_direct_path_futures(response_tuples, crowfly_stop_points, odt_stop_points)
-        for future in gevent.iwait(futures):
-            resp_key, resp_direct_path = future.get()
-            g.fallback_direct_path_pool[resp_key] = resp_direct_path
-
-        # Now we construct the whole journey by concatenating the fallback direct path with the pt journey
-        worker.build_journeys(response_tuples, crowfly_stop_points, odt_stop_points)
-
-        #If resp doesn't contain any response we have to add an error message
-        if len(resp) == 0:
-            if len(g.origins_fallback[dep_mode]) == 0 and len(g.destinations_fallback[arr_mode]) == 0:
-                resp.append(self._make_error_response("no origin point nor destination point",
-                                                      response_pb2.Error.no_origin_nor_destination))
-            elif len(g.origins_fallback[dep_mode]) == 0:
-                resp.append(self._make_error_response("no origin point", response_pb2.Error.no_origin))
-            elif len(g.destinations_fallback[arr_mode]) == 0:
-                resp.append(self._make_error_response("no destination point", response_pb2.Error.no_destination))
-            return resp
-        for r in resp:
+        #
+        for mode in requested_dep_modes:
+            direct_path_pool.add_async_request(requested_orig_obj, requested_dest_obj, mode,
+                                               fallback_extremity, request, direct_path_type)
+
+        # if max_duration(time to pass in pt) is zero, there is no need to continue,
+        # we return all direct path without pt
+        if not request.get('max_duration', 0):
```

Comment:
From what I understand of the intend, I'd write:
```python
if  request['max_duration'] == 0:
```
As if it's `None` (which probably won't be the case because default is 86400) we want to do a "complete" journeys computation.